erdos/src/dataflow/graph/graph.rs

```diff
@@ -0,0 +1,342 @@
+use std::sync::{Arc, Mutex};
```

Comment:
Can we move this file into `graph/mod.rs`? I know we do this in `node.rs`, but it's [bad style](https://rust-lang.github.io/rust-clippy/master/#module_inception).

Reply:
I pushed an update to the visibility of the modules, and I agree with the bad styling, but don't necessarily agree that it should be moved to `mod.rs`. The style guide just says that imports shouldn't be of the form `use graph::graph::Graph`, but we could have `pub(crate) graph` in `graph/mod.rs` and publicly export `Graph` from inside the second graph module. 

I feel like the graph module's `mod.rs` is being used for abstract operator and stream types and corresponding traits, which don't necessarily go well with placing the publicly-visible `Graph` abstraction with them.

Tokio seems to be doing something similar, although not exactly the same with their `stream_ext` module [here](https://github.com/tokio-rs/tokio/blob/master/tokio-stream/src/lib.rs).